src/deepsparse/utils/onnx.py

```diff
@@ -68,6 +68,49 @@
 }
 
 
+def save_onnx_to_path(model: Model, model_path: str, external_data_file: str) -> bool:
+    """
+    Save model to the given path.  If the model has external data, store the
+    external data in 'external_data_file'.
+    Returns False if the model had no external data, True otherwise.
+    """
+    try:
+        onnx.checker.check_model(model)
+        onnx.save(model, model_path)
+        return False
+    except ValueError:
+        # Presumably the checker has failed because of external data
```

Comment:
I think this is fine for now, we could also try checking the total size of the proto in memory to see if it's at or around 2G

Reply:
I found a check like this in the onnx source code so I've applied similar logic here.